src/app/games/gos/tri-go/tests/TriGoRules.spec.ts

```diff
@@ -0,0 +1,756 @@
+                [N, N, N, _, O, _, N, N, N],
+                [N, N, _, _, _, _, _, N, N],
+                [N, _, _, _, _, _, _, _, N],
+                [w, X, _, _, _, _, _, _, _],
+            ];
+            const expectedState: GoState =
+                new GoState(expectedBoard, PlayerNumberMap.of(1, 1), 4, MGPOptional.empty(), GoPhase.COUNTING);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState, defaultConfig);
+        });
+
+        it('simply shared board should be simple to calculate', () => {
+            // Given a simply shared board on which one player passed
+            const previousBoard: Table<GoPiece> = [
+                [_, _, O, X, _],
+                [_, _, O, X, _],
+                [_, _, O, X, _],
+                [_, _, O, X, _],
+                [_, _, O, X, _],
+            ];
+            const state: GoState =
+                new GoState(previousBoard, noCaptures, 10, MGPOptional.empty(), GoPhase.PASSED);
+
+            // When passing again
+            const move: GoMove = GoMove.PASS;
+
+            // Then board value should change to match simple territory
+            const expectedBoard: GoPiece[][] = [
+                [b, b, O, X, w],
+                [b, b, O, X, w],
+                [b, b, O, X, w],
+                [b, b, O, X, w],
+                [b, b, O, X, w],
+            ];
+            const expectedState: GoState =
+                new GoState(expectedBoard, PlayerNumberMap.of(10, 5), 11, MGPOptional.empty(), GoPhase.COUNTING);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState, defaultConfig);
+        });
+
+    });
+
+    describe('GoPhase.COUNTING', () => {
+
+        it('should be GameStatus.ONGOING', () => {
+            // Given a board with a shared territory
+            const board: Table<GoPiece> = [
+                [b, b, O, X, u],
+                [b, b, O, X, w],
+                [b, b, O, X, w],
+                [b, b, O, X, X],
+                [k, k, O, X, w],
+            ];
+            const state: GoState =
+                new GoState(board, PlayerNumberMap.of(10, 1), 0, MGPOptional.empty(), GoPhase.COUNTING);
+            const node: GoNode = new GoNode(state);
+
+            // When evaluating it
+            // Then it should be considered as ongoing
+            RulesUtils.expectToBeOngoing(rules, node, defaultConfig);
+        });
+
+        it('should attribute shared territory to surviving group', () => {
+            // Given a board with a shared territory
+            const board: Table<GoPiece> = [
+                [b, b, O, X, O],
+                [b, b, O, X, _],
+                [b, b, O, X, _],
+                [b, b, O, X, X],
+                [b, b, O, X, w],
+            ];
+            const state: GoState =
+                new GoState(board, PlayerNumberMap.of(10, 1), 0, MGPOptional.empty(), GoPhase.COUNTING);
+
+            // When marking as dead one of the sharing group
```

Comment:
use the same words: shared territory

Reply:
j'vais clarifier mais non, tu marque pas un territoire comme mort, c'pas d'bon sang